### ISSUE TITLE
twisted.plugin documentation errors

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2169]
|<img alt="babbitt's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| babbitt reported|
|-|-|
|Trac ID|trac#2169|
|Type|task|
|Created|2006-10-11 19:48:34Z|

RE:
http://twistedmatrix.com/projects/core/documentation/howto/plugin.html

The code example under the header: "Extending an Existing Program" does not indicate all imports that must be used.  In addition to the imports that are listed within, one also needs "from zope.interface import implements" or else the module will fail.

Attachments:

* [doc-errors-2169.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/f15/f157b1011d9fed3d73348ae4962471c017c95dfb/5a16238d54ef600857d85c293cbf74a48530e7e7.patch) (371 bytes) - added by collab on 2008-07-01 17:45:18Z - patch against r24199

<details><summary>Searchable metadata</summary>

```
trac-id__2169 2169
type__task task
reporter__babbitt babbitt
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1160596114000000 1160596114000000
changetime__1214951787000000 1214951787000000
version__None None
owner__ 
cc__collab
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph set owner to babbitt|
|-|-|

Would you mind contributing a patch?

**Comment by twisted-trac** [comment]
|<img alt="collab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|collab set owner to collab<br>collab set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="collab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|collab removed owner<br>collab set status to `new`|
|-|-|

Attached a [[doc-errors-2169.patch](None/doc-errors-2169.patch) patch] that fixes the import, ready for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

Third party imports should be placed above Twisted imports and below stdlib imports.  Re-arranging the diff, then applying. :)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [24208]) Apply doc-errors-2169.patch - add missing zope.interface import

Author: collab
Reviewer: exarkun
Fixes: [#2169](https://github.com/twisted/twisted/issues/2169)

An example in the plugin howto used `implements` from
`zope.interface` but did not import it; change this so
that readers know where it comes from.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
